run-checks

```diff
@@ -134,27 +142,22 @@ if [ "$STATIC" = 1 ]; then
     # check commit author/committer name for unicode
     ./check-commit-email.py
 
-    if [ -z "${SKIP_GOFMT:-}" ]; then
-        echo Checking formatting
-        fmt=""
-        for dir in $(go list -f '{{.Dir}}' ./... | grep -v '/\(c-\)\?vendor/' ); do
-            # skip vendor packages
-            s="$(${GOFMT:-gofmt} -s -d "$dir" || true)"
-            if [ -n "$s" ]; then
-                fmt="$s\\n$fmt"
-            fi
-        done
-        if [ -n "$fmt" ]; then
-            echo "Formatting wrong in following files:"
-            # shellcheck disable=SC2001
-            echo "$fmt" | sed -e 's/\\n/\n/g'
-            exit 1
-        fi
+    # golangci's output isn't as clear when there are compilation errors
+    go build github.com/snapcore/snapd/...
+
+    if ! command -v golangci-lint >/dev/null; then
+      go get -u github.com/golangci/golangci-lint
     fi
 
-    # go vet
-    echo Running vet
-    go list ./... | grep -v '/vendor/' | xargs go vet
+    if [ -n "${SKIP_GOFMT:-}" ]; then
+      # it's not possible to disable a specific linter through the CLI and have
+      # 'disable-all: true' in the config. This comments out gofmt in the config
+      sed -i 's/- gofmt/#- gofmt/' .golangci.yml
```

Comment:
I looked at golangci-lint source code and they really make it impractical. Looks like the configuration is merged, so our disable-all in the config is equal to --disable-all, then we selectively enable specific linters, like through --enable. Now, passing --disable gofmt actually triggers this code https://github.com/golangci/golangci-lint/blob/aa3550ab837926f70329a449adb4d826517016b1/pkg/lint/lintersdb/validator.go#L66-L68, but even if we fixed that, I believe it would fail next in https://github.com/golangci/golangci-lint/blob/aa3550ab837926f70329a449adb4d826517016b1/pkg/lint/lintersdb/validator.go#L84-L88

How about we leave gofmt outside of golangci-lint for now?

Reply:
Yes, I didn't expect that having disable-all to interfere with disabling a specific linter through the cli. This was a quick and dirty workaround but, I agree, it's not great. However, since this will also need to skip nakedret, I'll try to find a better way to do it instead of just leaving them out of golangci